xds/src/test/java/io/grpc/xds/internal/rbac/engine/GrpcAuthorizationEngineTest.java

```diff
@@ -177,6 +179,55 @@ public void headerMatcher() {
     assertThat(decision.decision()).isEqualTo(Action.DENY);
   }
 
+  @Test
+  public void headerMatcher_binaryHeader() {
+    AuthHeaderMatcher headerMatcher = new AuthHeaderMatcher(Matchers.HeaderMatcher
+        .forExactValue(HEADER_KEY + Metadata.BINARY_HEADER_SUFFIX, HEADER_VALUE, false));
+    OrMatcher principal = OrMatcher.create(headerMatcher);
+    OrMatcher permission = OrMatcher.create(
+        new InvertMatcher(new DestinationPortMatcher(PORT + 1)));
+    PolicyMatcher policyMatcher = new PolicyMatcher(POLICY_NAME, permission, principal);
+    GrpcAuthorizationEngine engine = new GrpcAuthorizationEngine(
+        new AuthConfig(Collections.singletonList(policyMatcher), Action.ALLOW));
+    Metadata metadata = new Metadata();
+    metadata.put(Metadata.Key.of(HEADER_KEY + Metadata.BINARY_HEADER_SUFFIX,
+        Metadata.BINARY_BYTE_MARSHALLER),
+        BaseEncoding.base64().encode(HEADER_VALUE.getBytes(US_ASCII)).getBytes(US_ASCII));
```

Comment:
This should not be encoded. Pass `HEADER_VALUE.getBytes(US_ASCII)`. gRPC handles the base64 encoding.